### DISCUSSION TITLE
Classified Positions in the Session Results

### DISCUSSION THREAD

**Discussion by CraigAT** [theOehrly/Fast-F1#150] [Ideas]
Hi.  
I hope you can help.  
I am trying to work with the session.results of this years races but I have hit a stumbling block when looking for the classified position of a driver.
This year's second race (the Saudi Arabian GP) is a good example.  The fastf1 session results show the driver "positions" are listed from HUL in 12th down to TSU in 19th, but the [F1 Official results](https://www.formula1.com/en/results.html/2022/races/1125/saudi-arabia/race-result.html) only classify places down to 14th place (being ALB).  I don't see an easy way, given the data in that dataframe to calculate whether a driver was classified or not (in other races you could assume the status of "finished" or "+x laps" implies a classification, but that doesn't explain ALB in this race).  In the docs I see a "dnf" column mentioned but I do not see that in the data.
I thought this may be an issue with the source data, but if you are using the Ergast API, then the [results there](http://ergast.com/api/f1/2022/2/results) carry the correct classification (and classification positions).
If the classified position information is available in the Fast-F1 package then please can you point me to the correct documentation or how to extract it, please?  If the information does not exist, please could it be added?  Perhaps as a "Classification" column, copied directly from the Ergast data.  Hopefully that would not break backwards compatibility and would allow us to keep the current "Position" column which is still useful for quick ranking or sorting of the results.
I hope I have explained the issue well enough, and that someone can suggest a solution.
Many Thanks
Craig

**Comment by markste-in** [comment]
Albon is considered classified because he completed over 90% of the entire race distance

You could manually implement the ruleset to derive the classification status but it would be better to use the official f1 results
<img width="686" alt="image" src="https://user-images.githubusercontent.com/33337590/162973973-3b9c7393-7e32-4682-a468-018c9be4101b.png">

I also don't see a classification flag in the ergast data. Maybe I am missing something?

<img width="742" alt="image" src="https://user-images.githubusercontent.com/33337590/162974455-537b0ead-2d31-4827-b94e-60acc657347d.png">

**Reply by CraigAT** [reply]
Hi.

Thanks for the quick reply and the awesome module!

I understood the requirements for classification and I apologise for the confusion (I agree there is not a "classification flag" in the Ergast data).  In the Ergast data the classification (matching the official F1 race classification) can be implied or is shown by the "position" field:

![image](https://user-images.githubusercontent.com/10949207/163067646-8ef6dcda-8bb9-4125-8d32-776603d7df71.png)


Whereas in the fastf1 session.results data it is not possible to deduce classification or classified position (from Albon down):

![image](https://user-images.githubusercontent.com/10949207/163067772-0725a503-a2ae-47da-9d20-2561c7a1b47f.png)

(Note. I have dropped some columns from the dataframe to concentrate on the relevant info)

It appears the fastf1 'position' is more of a finished ranking (which is still a useful field) rather an official 'classified position' as used in F1 and Ergast data.  This difference in use of the 'position' field may be by design, but was not what I was expecting.  I was hoping a 'classification' field could be added to the existing fields (including the current 'position' field) that would have the Ergast position values.

Kind regards.

Craig

**Reply by theOehrly** [reply]
Hi, I didn't really know about that classification rule. You are right, it'd be worth adding this information too. I'll see when I might get around to it, but don't expect anything soon.

If someone else wants to try, it shouldn't be too difficult to add. But the code is a bit convoluted, merging two data sources, depending on what data is available.
